### PR TITLE
Feat/cancel active subscription

### LINKED ISSUE #21: User should be able to cancel their subscription
**What:** User should be able to cancel their subscription

**Why:** For users be able to manage their subscription

**How:**

- The subscription section the back-office will have all the information about their user current subscription.
- There's a button called "Cancel subscription", if pressed will prompt for user confirmation before canceling the subscription.
- Follow react-admin conventions for style

### PR DESCRIPTION
**Pull request by castrolem**
**What:** User should be able to cancel their subscription

**Why:** For users to be able to manage their subscription
Closes #21 

**How:**
By adding a cancel subscription button in the users' profile view.

## Screenshots
image

### CODE REVIEW

File: app/javascript/bundles/User/components/UserShow.jsx [hunk 1]
```diff
@@ -16,24 +16,25 @@ function UserShowView ({ user, subscription, streak }) {
...
-      <p>{user.name}</p>
-
-      <p>
-        <strong>Email:</strong>
-      </p>
-      <p>{user.email}</p>
-
-      {subscription && streak && (
+    <>
+      {subscription && subscription.active && streak && (
```
Comment (orlando) [inline app/javascript/bundles/User/components/UserShow.jsx:20]: What about using safe navigation operator here?

```ruby
subscription&.active && streak
```

File: app/models/user.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class User < ApplicationRecord
 
   SSO_ATTRIBUTES = %w[admin avatar_url banned custom_fields email name username].freeze
 
-  has_one :subscription
+  has_one :subscription, -> { where active: true }
```
Comment (orlando) [inline app/models/user.rb]: Why is this needed?

File: config/routes.rb [hunk 3]
```diff
@@ -21,6 +21,7 @@
   resources :users, only: %i[show new edit update create destroy] do
     resource :streak, only: %i[show]
     resources :cards, only: %i[index destroy]
+    resource :subscription, only: %i[destroy]
```
Comment (orlando) [inline config/routes.rb:24]: @castrolem are we considering the case where users can update their subscription level? Or how that will work?

File: spec/features/user/users_profile_spec.rb [hunk 4]
```diff
@@ -27,5 +27,25 @@
...
+      expect(page).to have_content(subscription.plan.name)
+
+      click_button 'Cancel Subscription'
+
+      expect(page).to have_content('Do you want to terminate your current subscription?')
+      within '#cancel-subscription-dialog' do
+        click_button 'Cancel Subscription'
+      end
+
+      sleep 2
```
Comment (orlando) [inline spec/features/user/users_profile_spec.rb]: @castrolem this will probably lead to flaky tests, can we find a better approach?

**Review by orlando — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (castrolem) [reply app/models/user.rb]: Because you can have a previous plan that's inactive, and get a new one that's different from that one that will be active.

File: config/routes.rb (hunk 3 above)
Comment (castrolem) [reply config/routes.rb:24]: We're not, right now you can only cancel to get the other plan.

Comment (castrolem) [reply config/routes.rb:24]: Since there can only be 1 plan active for a user at a time.

Comment (orlando) [reply config/routes.rb:24]: ok if this is the case, then a user should be able to have many subscriptions but only one should be active at a time. Should we change the model to reflect this? This is also related to my previous comment https://github.com/debtcollective/fundraising/pull/107#discussion_r334593067

File: app/models/user.rb (hunk 2 above)
Comment (orlando) [reply app/models/user.rb]: Ok, see my comment below

File: app/javascript/bundles/User/components/UserShow.jsx (hunk 1 above)
Comment (castrolem) [reply app/javascript/bundles/User/components/UserShow.jsx:20]: We don't have it in JavaScript (yet)

File: config/routes.rb (hunk 3 above)
Comment (castrolem) [reply config/routes.rb:24]: Sounds good, let's make the changes on the model level.

File: app/javascript/bundles/User/components/UserShow.jsx (hunk 1 above)
Comment (orlando) [reply app/javascript/bundles/User/components/UserShow.jsx:20]: right!, I thought this was ruby for a second 😅

**Review by orlando — APPROVED** [review]
LGTM 👍, let's fix the failing test case and merge @castrolem
